src/Orchard.Cms.Web/Modules/Orchard.Roles/Services/RoleStore.cs

```diff
@@ -58,10 +58,10 @@ public void UpdateRoles(RolesDocument roles)
             _memoryCache.Set(Key, roles);
         }
 
-        public async Task<IEnumerable<string>> GetRoleNamesAsync()
+        public async Task<IEnumerable<RoleNamesEntry>> GetRoleNamesAsync()
```

Comment:
It is less optimal because we will read roles document for each role but it makes sense and performance in this scenario is not critical. I will do it.